<!DOCTYPE html>
<html>
  <head>
    <title>slncky: a tool for lncRNA discovery<</title>
    <link rel="stylesheet" type="text/css" href="https://bootswatch.com/paper/bootstrap.min.css">
  </head>

  <body>
    <nav class="navbar navbar-inverse">
      <div class="container-fluid" style="position: absolute; width: 100%;">
        <div class="navbar-header">
          <a class ="navbar-brand" href="index.html"><p style="font-size:26"><em>slncky</em> : a tool for lncRNA discovery</p></a>
        </div >
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-2">
          <ul class="nav navbar-nav">
          </ul>
          <ul class="nav navbar-nav navbar-right">
            <li><a href="install.html">Installation</a></li>
            <li class="active"><a href="#">Usage</a></li>
            <li><a href="cite.html">Cite</a></li>
            <li><a href="contact.html">Contact</a></li>
          </ul>
        </div>
      </div>
    </nav>

    <div class="container">
      <div class="jumbotron">
        <h2>Usage</h2>
        <hr>
          This page will teach you how to:
          <ul>
            <li><a href="#step-3-run-slncky">Run</a> <em>slncky</em></li>
            <li><a href="#customize">Customize</a> <em>slncky</em></li>
            <li>Understand <em>slncky</em>'s <a href="#output">output</a></li>
          </ul>
          <br>
          
        <h4>
        <a name="step-3-run-slncky" class="anchor" href="#step-3-run-slncky"><span class="octicon octicon-link"></span></a>Run <em>slncky</em> !</h4>
        <hr>
        <p>Usage: <code>./slncky.v2.0 bedfile assembly out_prefix</code></p>
        <div class="panel panel-default">
  <div class="panel-body">
        <h5>Examples</h5>
        <br>barebones:
        <br><code>./slncky.v2.0 transcripts.bed mm9 out</code>
        <br><br>don't filter, just search for orthologs:
        <br><code>./slncky.v1.0 -2 transcripts.bed mm9 out</code>
        <br><br>by default, <em>slncky</em> uses the <code>annotations.config</code> file in the same directory as itself, but you can set this path if you want:
        <br><code>./slncky.v1.0 -c ./annotations.config transcripts.bed mm9 out</code>

        <br><br>if bedtools is not in your path:
        <br><code>./slncky.v1.0 --bedtools_path /unix/home/jjenny/bedtools-2.17.0/bin transcripts.bed mm9 out</code>
</div></div>

        <a name="customize" class="anchor" href="#customize"><span class="octicon octicon-link"></span></a>
        <table class="table table-striped table-hover ">
        <thead>
        <tr><td colspan="2"><h4>Customize <em>slncky</em></h4></td></tr>
        </thead>
        <tbody>
        <tr><td><h6>General Options</h6></td><td></td></tr>    
        <tr><td><code>-c</code> / <code>--config</code></td><td>Specify path of <code>annotations.config</code> file. (Default:<code>annotations.config</code> assumed to be in same directory as <em>./slncky</em> executable.)</td></tr>
        <tr><td><code>-1 </code>/ <code>--no_orth_search</code></td><td>Only filter; don't search for orthologs.</td></tr>
        <tr><td><code>-2 </code>/ <code>--no_filter</code></td><td>Don't filter; only search for orthologs.</td></tr>
        <tr><td><code>-o </code>/ <code>--overwrite</code></td><td>Force overwrite of trancsripts.lncs.bed.</td></tr>
        <tr><td><code>-n </code>/ <code>--threads</code></td><td>Number of threads to use for aligning transcripts. (Default: 5)</td></tr>
        <tr><td><code>-w </code>/ <code>--web</code></td><td>Create browser to visualize aligned lncRNAs.</td></tr>
        <tr><td><code>--bedtools</code></td><td>Path to <em>bedtools</em> executables. (Default: bedtools assumed to be in user's PATH)</td></tr>
        <tr><td><code>--liftover</code></td><td>Path to <em>liftover</em> executable. (Default: bedtools assumed to be in user's PATH)</td></tr>
        <tr><td><code>--lastz</code></td><td>Path to <em>lastz</em> executable. (Default: bedtools assumed to be in user's PATH)</td></tr>

        <tr><td><h6>Filter Options</h6></td><td></td></tr>
        <tr><td><code>--min_overlap</code></td><td>Minimum overlap for filtering transcripts overlapping <b>CODING</b> or <b>MAPPED_CODING</b> gene. (Default: 0.00)</td></tr>
        <tr><td><code>--min_cluster</code></td><td>Minimum size of duplication cluster for filtering paralogous transcripts. (Default: 2)</td></tr>
        <tr><td><code>--min_coding</code></td><td>Minimum identity for filtering transcripts that align to syntenic <b>CODING</b> gene. (Default: <em>slncky</em> learns threshold from input data.)</td></tr>
        <tr><td><code>--no_collapse</code></td><td>Do not collapse overlapping transcripts into single canonical gene.</td></tr>
        <tr><td><code>--no_overlap</code></td><td>Do not remove transcripts that overlap <b>CODING</b> or <b>MAPPED_CODING</b> gene.</td></tr>
        <tr><td><code>--no_self</code></td><td>Do not remove transcripts sharing sequence similarity with each other.</td></tr>
        <tr><td><code>--no_dup</code></td><td>Do not remove transcripts that share sequence similarity with <b>DUP</b> genes.</td></tr>
        <tr><td><code>--no_coding</code></td><td>Do not remove transcripts that align to syntenic <b>CODING</b> genes.</td></tr>
        <tr><td colspan="2"><h6>Orthology Search Options</h6></td></tr>
        <tr><td><code>--min_noncoding</td><td>Minimum overlap for reporting transcripts that align to syntenic <b>NONCODING</b> gene.  (Default: <em>slncky</em> learns threshold by aligning to shuffled intergenic regions)</td></tr>
        <tr><td><code>--no_bg</td><td>Do not align to shuffled background; report all alignments.  Setting this option will greatly reduce <em>slncky</em> runtime.</td></tr>
        <tr><td><code>--no_orf</td><td>Do not check for conserved ORFs in aligned noncoding transcripts.</td></tr>
        <br><br>
        <tr><td colspan="2"><h6>Advanced orthology search options</h6></td></tr>
        <tr><td><code>--minMatch</td><td>minMatch parameter to pass to <em>liftOver</em>.  Decreasing this parameter will increase sensitivity of orthology search. (Default: 0.1)</td></tr>
        <tr><td><code>--pad</td><td>Number of basepairs up- and downstream of lncRNA loci to include in orthology search. Increasing this parameter will increase sensitivity of orthology search. (Default: 50,000)</td></tr>
        <tr><td><code>--gap_open</td><td>Gap open penalty to pass to <em>lastz</em>. Decreasing this parameter will increase sensitivity of orthology search. (Default: 200)</td></tr>
        <tr><td><code>--gap_extend</td><td>Gap extend penalty to pass to <em>lastz</em>. Decreasing this parameter will increase sensitivity of orthology search. (Default: 40)</td></tr>
        <tr><td></td><td></td></tr>
       </tbody>
       </table> 

<br>
<h4><a name="output" class="anchor" href="#output"><span class="octicon octicon-link"></span></a>Output</h4>
<hr>
<code>out.lncs.bed</code>
<br>Bed12 file of high-confidence long, non-coding RNAs
<br><br><code>out.filtered_info.txt</code> / <code>out.cluster_info.txt</code>
<br>filter_info contains a list of every filtered out transcripts with the reason for filtering. cluster_info contains names of transcripts found in duplication clusters
<br><br><code>out.canonical_to_lncs.txt</code>
<br>A table of the canonical transcripts chosen from groups over overlapping transcripts.
<br><br><code>out.orthologs.txt</code> / <code>out.orthologs.top.txt</code>
<br>A list of every ortholog pair, with additional information about the alignment. The out.orthologs.top.txt file contains only the best alignment (based on exonic identity) for each lncRNA.
<br><br><code>out.orfs.txt</code>
<br>A list of alignments which contain ORFs longer than 30bp, along with additional information about the ORF alignment.</p> 



      </div>
    </div>

  
  </body>
</html>
